demo/site/src/util/createSiteUrl.ts

```diff
@@ -0,0 +1,33 @@
+type CreateSiteUrlOptions = {
+    /**
+     * path with leading /
+     *
+     * Sample: "/test/to/my/page"
+     */
+    path: string;
+
+    scope: {
+        language: string;
+    };
+
+    /**
+     * If provided, the anchor will be appended to the resulting URL
+     *
+     * Sample: "my-anchor"
+     */
+    anchor?: string;
+};
+/**
+ * Creates a URL from the provided options.
+ *
+ * @return {string} The resolved URL: /{scope.language}/test/to/my/page#my-anchor
+ */
+export const createSiteUrl = ({ path, scope, anchor }: CreateSiteUrlOptions) => {
```

Comment:
I'd name this function createPath or createSitePath, since it isn't really the URL.

Reply:
It is a relative Url. (And also a Path)

The important thing is to include _Site_ in the function name, as it is the Url (Or path) used by the site. (vs. api - which doesn't know about the language prefix)